### PR TITLE
oneauth authentication using cb api

### LINKED ISSUE #2: Setup authentication using oneauth
Currently, the project uses Django authentication mechanism but we need to authenticate users via our SSO platform [oneauth](https://account.codingblocks.com) 

To get started with this task:
1. I'll recommend you to read about the authentication flow in OAuth 2.0. You refer [this](https://www.digitalocean.com/community/tutorials/an-introduction-to-oauth-2) article.
2. Go to https://account.codingblocks.com and create a new Client under My Clients tab
    - Use localhost:8000 as the callback url and default url
3.  Create two vars in `settings.py` file `ONEAUTH_CLIENT_ID` and `ONEAUTH_CLIENT_SECRET` whose values are fetched from the .env file

ToDos for the task are:
- [ ] Replace `LOGIN_URL` in the `settings.py` with the authorization URL of oneauth
- [ ] Create a new application named `oneauth` and an endpoint which handles the callback from oneauth
- [ ] Exchange grant code with the access token
- [ ] Use access token to retrieve the user detail from the oneauth using `/api/users/me` endpoint
- [ ] Upsert user from the database
    - Use email id to fetch the user from database
        - [ ] If user exists update the user details and log in the user
        - [ ] If user doesn't extsts create the user and log in

### PR DESCRIPTION
**Pull request by tushargoyal22**
Fixes #2 

- [x] Replace LOGIN_URL in the settings.py with the authorization URL of oneauth
- [x]  Create a new application named oneauth and an endpoint which handles the callback 
- [x] Exchange grant code with the access token
- [x] Use access token to retrieve the user detail from the oneauth using /api/users/me endpoint
- [x] Upsert user from the database
     - Use email id to fetch the user from database
       - [x]  If user exists update the user details and log in the user
       - [x]  If user doesn't extsts create the user and log in

### CODE REVIEW

**Comment by tushargoyal22** [comment]
Bhaiya, I had also already done but didn't make PR because waiting for 15, same happened for issue1

**Comment by tushargoyal22** [comment]
@jatinkatyal13   I had completed this in evening only completely. but was waiting. So for future please add PR and contribution rules, and suggest me to proceed with this

**Comment by tushargoyal22** [comment]
@jatinkatyal13 please review the PR

File: poker/settings.py [hunk 1]
```diff
@@ -63,13 +63,20 @@
 # Database
 # https://docs.djangoproject.com/en/3.0/ref/settings/#databases
 
+# DATABASES = {
```
Comment (jatinkatyal13) [inline poker/settings.py:66]: We need to use the postgres in development environment as well.

File: poker/urls.py [hunk 2]
```diff
@@ -1,9 +1,17 @@
...
 from django.urls import path
 from services import crontab
+from django.conf.urls import url
+from django.contrib.auth import views as auth_views
+from django.views.generic.base import TemplateView
+from oneauth import views
 
 urlpatterns = [
     path('admin/', admin.site.urls),
+    url(r'^$', TemplateView.as_view(template_name='home.html'), name='home'),
```
Comment (jatinkatyal13) [inline poker/urls.py:11]: let's use the path function for creating the url routes

File: poker/settings.py (hunk 1 above)
Comment (tushargoyal22) [reply poker/settings.py:66]: Fine , actually on my system I was facing some issues with postgres. I will correct this

Comment (tushargoyal22) [reply poker/settings.py:66]: now I need to close this pr? as already this issue is merged by another PR

**Pull request closed without merging**